Ingame Sound Fxs and Background Music Fix for Scene Transition

Eriksen2411 opened the pull request:
-  Add all necessary sound effects for the game (BlockHit, Die, Dodge, Hurt, Revive, Attack, Ready Attack)

- Create duplicate of existing prefabs and add the SoundFXIndexLibrary to differentiate between characters sound (since we are using same Combat scripts for all).
- Tested all sound effects with level 3 (duplicate of level 1 but change in prefabs used)
- Fix the problem from previous pr when coming back to menu, the background music is not changed successfully.

- Please test the sound effects by loading level 3 instead of level 1.
- After merged this pr, can put the pr change on top of the sprites-changed prefabs and delete the duplicate prefabs.
- Notice:
+ There is only 1 way to enter the levels from menus is through the level select scene, which is dealt with already.
+ To support change of background music correctly from ingame to other menu scenes, please add the following lines when you have any changes from levels to menus:

AudioManager.Instance.StopMusic(Music.LibraryIndex.INGAME_BACKGROUND_MUSIC);
AudioManager.Instance.PlayMusic(Music.LibraryIndex.MENU_BACKGROUND_MUSIC);

Background music credits:
Now We Ride by Alexander Nakarada | https://www.serpentsoundstudios.com
Music promoted by https://www.chosic.com/free-music/all/
Creative Commons CC BY 4.0
https://creativecommons.org/licenses/by/4.0/

Closes #44.

Eriksen2411 committed:
-  Add all necessary sound effects for the game (BlockHit, Die, Dodge, Hurt, Revive, Attack, Ready Attack)
- Create duplicate of existing prefabs and add the SoundFXIndexLibrary to differentiate between characters sound (since we are using same Combat scripts for all).
- Tested all sound effects with level 3 (duplicate of level 1 but change in prefabs used)
- Fix the problem from previous pr when coming back to menu, the background music is not changed successfully.

- Please test the sound effects by loading level 3 instead of level 1.
- After merged this pr, can put the pr change on top of the sprites-changed prefabs and delete the duplicate prefabs.
- Notice:
+ There is only 1 way to enter the levels from menus is through the level select scene, which is dealt with already.
+ To support change of background music correctly from ingame to other menu scenes, please add the following lines when you have any changes from levels to menus:

AudioManager.Instance.StopMusic(Music.LibraryIndex.INGAME_BACKGROUND_MUSIC);
AudioManager.Instance.PlayMusic(Music.LibraryIndex.MENU_BACKGROUND_MUSIC);

Eriksen2411 committed:
Merge branch 'master' into danh_sound

Eriksen2411 committed:
- Add the knight jump sound fx, level entry fx.
- Change several audio for sound fxs.
- Apply the changes on prefabs and level 1, delete duplicate prefabs and level 3.

lowkangn reviewed (approved):
LGTM, but there are a few issues:
The sounds for the Dragon's attack and Dragon enemy death cut out abruptly, advise adding a fade-out.
The Knight's jumping sound seems to play only after he lands on the ground and not when he starts jumping. Is this intended?
Consider adding a sound when the Knight. Due to the new sprite it can be hard to determine if the Knight was buffed or not.

Otherwise, great job on the sound selection! I really like the sounds for the Dragon's dash and the lock-on sounds for the ranged enemies.

Eriksen2411 committed:
- Fix Jump Delay Issue as stated.
- Change audio to avoid abrupt cut out of soundfx for dragon attack and dragon enemies' death.
